Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  AnnotationStrategies.swift
+//
+//
+//  Created by O'Brien, Patrick on 3/2/21.
+//
+
+import ARKit
+import Foundation
+import RealityKit
+import SwiftUI
+
+/// A loading strategy that uses the RealityComposer app. After creating the Reality Composer scene, tthe entities in the scene correlate to a real world location relative to the image or object anchor.
+/// This strategy wraps the anchors that represents these locations with the CardItemModels that they correspond to in a ScreenAnnotation struct for a single source of truth.
+/// Loading the data into the ARAnnotationViewModel should be done in the onAppear method.
+///
+/// - Parameters:
+///  - cardContents: An array of **CardItem : `CardItemModel`** which represent what will be displayed in the default CardView
+///  - anchorImage: Image to be converted to ARReferenceImage and added to ARConfiguration for discovery, can be nil if detecting an object Anchor
+///  - physicalWidth: The width of the image in meters
+///  - realityFileURL: URL path to a .reality file that makes contains the scene, exported from Reality Composer
+///  - sceneName: Name given to the scene in the Reality Composer app.
+///
+/// ## Usage
+/// ```
+/// let cardItems = [ExampleCardItem(id: 0, title_: "Hello"), ExampleCardItem(id: 1, title_: "World")]
+/// guard let anchorImage = UIImage(named: "qrImage") else { return }
+/// let realityFilePath = FileManager.default.getDocumentsDirectory().appendingPathComponent(FileManager.realityFiles).appendingPathComponent("ExampleRC.reality")
+/// let strategy = RealityFileStrategy(cardContents: cardItems, anchorImage: anchorImage, physicalWidth: 0.1, realityFilePath: realityFilePath, rcScene: "ExampleScene")
+/// arModel.load(loadingStrategy: strategy)
+/// ```
+public struct RealityFileStrategy<CardItem: CardItemModel>: AnnotationLoadingStrategy where CardItem.ID: LosslessStringConvertible {
```

Comment:
Looks like duplicate code as most of it is present in `RealityComposerStrategy`. Only main difference is the `init` and different use of `RCScannner.load...` function. Recommend to consolidate reusable code across strategies

Reply:
That is definitely fair. By consolidate, do you mean this should all just live within the RealityComposerStrategy with perhaps more inits and methods? There will be another URL which uses a usdz file that will be almost the exact same as this one as well.

Reply:
There can be different strategy implementation but internally we should share reusable code. I favor composition over inheritance